src/info/info_redis.c

```diff
@@ -270,3 +280,44 @@ void AddToInfo_RSConfig(RedisModuleInfoCtx *ctx) {
   RedisModule_InfoAddFieldLongLong(ctx, "min_phonetic_term_length",
                                    RSGlobalConfig.minPhoneticTermLen);
 }
+
+void AddToInfo_CurrentThread(RedisModuleInfoCtx *ctx) {
+  SpecInfo *specInfo = CurrentThread_TryGetSpecInfo();
+  if (!specInfo) {
+    return;
+  }
+  RedisModule_InfoAddSection(ctx, "current_thread");
+  if (specInfo) {
+    StrongRef strong = WeakRef_Promote(specInfo->specRef);
+    IndexSpec *spec = StrongRef_Get(strong);
+    if (!spec) {
+      RedisModule_InfoAddFieldCString(ctx, "index", specInfo->specName ? specInfo->specName : "n/a");
+    } else {
+      RedisModule_InfoAddFieldCString(ctx, "index", spec->name);
+      // output FT.INFO
+    }
+  }
+}
+
+void AddToInfo_ActiveQueries(RedisModuleInfoCtx *ctx) {
+  ActiveQueries *activeQueries = GetActiveQueries();
+  // If we are not the main thread then do not output the current queries
+  if (!activeQueries) {
+    return;
+  }
+  RedisModule_InfoAddSection(ctx, "active_queries");
```

Comment:
I think we should have the same structure always, and have empty sections in case there is nothing to print for that section, such that we will have a consistent format.